src/lib/bottom-sheet/bottom-sheet-config.ts

```diff
@@ -50,6 +50,14 @@ export class MatBottomSheetConfig<D = any> {
   /** Whether the bottom sheet should focus the first focusable element on open. */
   autoFocus?: boolean = true;
 
+  /**
+   * The focusable item to restore focus to if attempting restoreFocus is true.  Can be an object
+   * that has a focus method, or a function that returns this type of object.  If no object is
+   * provided the bottom sheet will attempt to return the focus to the element which had focus when
+   * when the bottom sheet opened.
+   */
+  focusRestoreTarget?: (() => {focus: () => any}) | {focus: () => any};
```

Comment:
Wouldn't this be somewhat easy and more flexible for consumers to implement on their own? E.g. this is the same as doing:

```ts
bottomSheet.open(SomeComponent, {restoreFocus: false}).afterClosed().subscribe(() => {
  someEl.focus();
});
```

The problem with this approach is that it's not obvious to the user that setting `restoreFocus` to `false` will also negate this option.

Reply:
That's a good point- when Joey and I were talking about this I hadn't considered using the `afterClose`.